elastic/kibana
# [Fleet] Don't error on missing package_assets value

## Summary
closes https://github.com/elastic/kibana/issues/89111

 * Update TS type to make `package_assets` key in EPM packages saved object optional
 * Update two places in code to deal with optional vs required property

### Checklist


#### Manual testing
1. checkout `7.10` branch
   1. **start ES:** `nvm use; yarn kbn bootstrap; yarn es snapshot --version 7.10.3 --license=trial  -E xpack.security.authc.api_key.enabled=true -E path.data=../data`
   1. **start Kibana**: `yarn start --no-base-path`
   1. **run** `curl -X POST -H 'kbn-xsrf: 1234' --user elastic:changeme localhost:5601/api/fleet/setup`
   2. **observe** `{"is_initialized: true}`
1. checkout `7.11` branch
   1. **start ES:** `nvm use; yarn kbn bootstrap; yarn es snapshot --version 7.11.1 --license=trial  -E xpack.security.authc.api_key.enabled=true -E path.data=../data`
   1. **start Kibana**: `yarn start --no-base-path`
   1. **run** `curl -X POST -H 'kbn-xsrf: 1234' --user elastic:changeme localhost:5601/api/fleet/setup`
   1. **observe** `{"is_initialized: true}`
1. checkout `master` branch
   1. **start ES:** `nvm use; yarn kbn bootstrap; yarn es snapshot --version 8.0.0 --license=trial  -E xpack.security.authc.api_key.enabled=true -E path.data=../data`
   1. **start Kibana**: `yarn start --no-base-path`
   1. **run** `curl -X POST -H 'kbn-xsrf: 1234' --user elastic:changeme localhost:5601/api/fleet/setup`
   1. **observe error** {"statusCode":500,"error":"Internal Server Error","message":"Cannot read property 'map' of undefined"}
1. checkout this PR `8911-fleet-startup-error`
   1. **start ES:** `nvm use; yarn kbn bootstrap; yarn es snapshot --version 8.0.0 --license=trial  -E xpack.security.authc.api_key.enabled=true -E path.data=../data`
   1. **start Kibana**: `yarn start --no-base-path`
   1. **run** `curl -X POST -H 'kbn-xsrf: 1234' --user elastic:changeme localhost:5601/api/fleet/setup`
   1. **observe success** `{"is_initialized: true}`

**_Notes_**
 * _you might need to do a `yarn kbn clean` when starting kibana if it fails. There have been some big changes in the tooling recently_
